nrf52840-hal/src/ieee802154.rs

```diff
@@ -0,0 +1,804 @@
+//! IEEE 802.15.4 radio
+
+use core::{
+    ops::{self, RangeFrom},
+    sync::atomic::{self, Ordering},
+};
+
+use embedded_hal::timer::CountDown as _;
+
+use crate::{
+    clocks::{Clocks, ExternalOscillator},
+    pac::{
+        generic::Variant,
+        radio::{state::STATE_A, txpower::TXPOWER_A},
+        RADIO,
+    },
+    timer::{self, Timer},
+};
+
+/// IEEE 802.15.4 radio
+pub struct Radio<'c> {
+    radio: RADIO,
+    // RADIO needs to be (re-)enabled to pick up new settings
+    needs_enable: bool,
+    // used to freeze `Clocks`
+    _clocks: &'c (),
+}
+
+/// Default Clear Channel Assessment method = Carrier sense
+pub const DEFAULT_CCA: Cca = Cca::CarrierSense;
+
+/// Default radio channel = Channel 20 (`2_450` MHz)
+pub const DEFAULT_CHANNEL: Channel = Channel::_18; // _17 todo undo; set this to interfere with my wifi
```

Comment:
It looks like this needs to be reset to channel 17.

Are the defaults specified somewhere or were they picked arbitrarily?

Reply:
the FREQUENCY register reset value is `2`, which doesn't match to any of the `Channel` values because they are all multiples of 5.
The channel values (the frequencies) are the standard IEEE frequencies.

Reply:
changed the default channel to 11 (2405 MHz) which is closer to the reset value of the FREQUENCY register (2402 MHz)